src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -406,6 +406,10 @@ public void deleteDataset(Integer datasetId, Integer userId) throws Exception {
     }
 
     private boolean filterDatasetOnProperties(DatasetDTO dataset, String term) {
+        //datasets need to have consentId, null check to prevent NPE
+        if(Objects.isNull(dataset.getConsentId())) {
+            return false;
+        }
         String consentId = dataset.getConsentId();
         return dataset.getProperties()
         .stream()
```

Comment:
If you return false on 411, then you'll never do any checking of the properties. I think you want to just ignore it if it isn't there, not return early.